lib/travis/build/addons/apt.rb

```diff
@@ -126,7 +126,7 @@ def add_apt_packages
 
             unless whitelisted.empty?
               sh.export 'DEBIAN_FRONTEND', 'noninteractive', echo: true
-              sh.cmd "sudo -E apt-get -yq update &>> ~/apt-get-update.log", echo: true, timing: true
+              sh.cmd "sudo -E apt-get -yq update", echo: true, timing: true
```

Comment:
It does return a non-zero value if there is an error, but not if there are only warnings.

Failing to resolve the repository host only generates a warning (0).
 > W: Some index files failed to download. They have been ignored, or old ones used instead.

Providing an invalid distribution name to a reachable repository host results in an error (100).
 > E: Some index files failed to download. They have been ignored, or old ones used instead.

The both have the same final message, but one is a warning and the other is an error.